sqlx-macros-core/src/query/data.rs

```diff
@@ -152,21 +152,34 @@ where
 
     pub(super) fn save_in(&self, dir: impl AsRef<Path>) -> crate::Result<()> {
         let path = dir.as_ref().join(format!("query-{}.json", self.hash));
-        let mut file = atomic_write_file::AtomicWriteFile::open(&path)
-            .map_err(|err| format!("failed to open the temporary file: {err:?}"))?;
+        match std::fs::remove_file(&path) {
```

Comment:
I thought this might fail on Windows if another process has the file open for reading, but Rust sets the `FILE_SHARE_DELETE` flag by default: https://doc.rust-lang.org/stable/std/os/windows/fs/trait.OpenOptionsExt.html#tymethod.share_mode

Reply:
Probably to mirror Unix behavior

Reply:
It appears that it can still result in errors, though. I created a test program that should simulate pretty well the different possible race conditions: https://gist.github.com/abonander/d083971d90e197c31711e06acf021461

It consistently fails with `PermissionDenied` on windows for `Test: 1, 1` (threads synchronize after the `fs::remove_file()` call)

Reply:
I suppose suppressing `PermissionDenied` from this call should be sufficient, though.

Reply:
Suppressing `PermissionDenied` from the `fs::remove_file()` call appears to fix it.

Reply:
https://gist.github.com/abonander/d083971d90e197c31711e06acf021461/49344746fe176fe3611197f3e09d44efaf5521a2